src/server/wtr-config.js

```diff
@@ -0,0 +1,157 @@
+import { argv } from 'node:process';
+import { defaultReporter } from '@web/test-runner';
+import { playwrightLauncher } from '@web/test-runner-playwright';
+
+const DEFAULT_PATTERN = type => `./test/**/*.${type}.js`;
+const DEFAULT_VDIFF = false;
+const ALLOWED_BROWSERS = ['chromium', 'firefox', 'webkit'];
+
+export class WTRConfig {
+
+	static #singleton;
+	#cliArgs = [];
+
+	constructor() {
+		if (!this.constructor.#singleton) {
+			this.cliArgs = argv;
+			this.constructor.#singleton = this;
+		}
+		return this.constructor.#singleton;
+	}
+
+	get #requestedBrowsers() {
+		return this.#cliArgs.toString().match(new RegExp(ALLOWED_BROWSERS.join('|'), 'gi'));
+	}
+
+	get visualDiffGroup() {
+		return {
+			name: 'vdiff',
+			files: this.pattern('vdiff'),
+			browsers: this.getBrowsers(['chromium']),
+			testRunnerHtml: testFramework =>
+				`<html lang="en">
+					<head>
+						<link rel="preload" href="https://s.brightspace.com/lib/fonts/0.5.0/assets/Lato-400.woff2" as="font" type="font/woff2" crossorigin>
+						<link rel="preload" href="https://s.brightspace.com/lib/fonts/0.5.0/assets/Lato-700.woff2" as="font" type="font/woff2" crossorigin>
+						<style>
+							html {
+								font-size: 20px;
+							}
+							body {
+								background-color: #ffffff;
+								margin: 0;
+								padding: 30px;
+							}
+							body[data-theme="dark"] {
+								background-color: #000000;
+							}
+							body[data-theme="translucent"] {
+								background: repeating-linear-gradient(45deg, #606dbc, #606dbc 10px, #465298 10px, #465298 20px);
+							}
+						</style>
+						<script type="module" src="./components/typography/typography.js"></script>
+					</head>
+					<body class="d2l-typography">
+						<script type="module" src="${testFramework}"></script>
+					</body>
+				</html>`
+		};
+	}
+
+	set cliArgs(cliArgs) {
+		this.#cliArgs = cliArgs;
+		this.requestedBrowsers = this.#requestedBrowsers;
+	}
+
+	create({
+		pattern = DEFAULT_PATTERN,
+		vdiff = DEFAULT_VDIFF,
+		timeout,
+		...passthroughConfig
+	} = {}) {
+		delete passthroughConfig.browsers;
+		this.pattern = pattern;
+
+		if (typeof pattern !== 'function') throw new TypeError('pattern must be a function');
+
+		const timeoutConfig = {};
+
+		if (typeof timeout !== 'undefined') {
+			if (typeof timeout !== 'number') throw new TypeError('timeout must be a number');
+
+			timeoutConfig.testFramework = {
+				config: {
+					timeout: timeout.toString()
+				}
+			};
+		}
+
+		const defaultConfig = {
+			files: pattern('test'),
+			nodeResolve: true,
+			testRunnerHtml: testFramework =>
+				`<html lang="en">
+					<body>
+						<script src="./tools/resize-observer-test-error-handler.js"></script>
```

Comment:
Done locally

Reply:
In your other PR, can we update core's version to just import this one? Then we're back to 1 copy. :)

Reply:
Clever! So I guess we'd temporarily export it? Which, speaking of, we probably need to add `/tools` anyways to https://github.com/BrightspaceUI/testing/blob/main/package.json#L19-L21?

Reply:
Or would this have worked without adding it, because our export needs it? I don't really know how `export` works, compared to regular public `files`

Reply:
We definitely need to add it to `files` in `package.json`, or move it under `src`. For exporting... yeah I'm not sure we might have to experiment. Since it's just being included as `<script src>` maybe whatever logic that prevents folks from importing things we don't export is bypassed.